gossipd/gossip.c

```diff
@@ -1552,12 +1553,13 @@ static struct io_plan *connection_in(struct io_conn *conn, struct daemon *daemon
 				   init_new_peer, daemon);
 }
 
-static void setup_listeners(struct daemon *daemon, u16 portnum)
+static void setup_listeners(struct daemon *daemon, u16 portnum, u8 *localsocket)
```

Comment:
`localsocket` is only ever read in this function, and is effectively an input string.  Maybe better to write as `const char *` and move the type conversion from `u8*` to `char *` to caller of `setup_listeners`.  Or at least `const u8*` in the argslist.